packages/rpc/src/__tests__/HttpClient.test.ts

```diff
@@ -908,6 +917,70 @@ describe(HttpClient, () => {
           "source_chain_expiry_block": 1n,
         }
       `);
+      expect(callCounter).toEqual(1);
+    });
+
+    it('handles multiple requests with 1 call', async () => {
+      const [r1, r2] = await Promise.all([
+        client.sendRequest(
+          'cf_pool_orders',
+          { chain: 'Ethereum', asset: 'ETH' },
+          { asset: 'USDC', chain: 'Ethereum' },
+        ),
+        client.sendRequest(
+          'broker_requestSwapDepositAddress',
+          { asset: 'BTC', chain: 'Bitcoin' },
+          { asset: 'ETH', chain: 'Ethereum' },
+          '0x4567',
+          100,
+          { gas_budget: '0x0', message: '0x0' },
+          30,
+          [{ account: '0x1234', bps: 0 }],
+        ),
+      ]);
+      expect(r1).toMatchSnapshot();
+
+      expect(r2).toMatchInlineSnapshot(`
+        {
+          "address": "0x1234",
+          "channel_id": 1,
+          "channel_opening_fee": 0n,
+          "issued_block": 1,
+          "source_chain_expiry_block": 1n,
+        }
+      `);
+      expect(callCounter).toEqual(1);
+    });
+
+    it('handles multiple requests separately if they are sent far apart', async () => {
+      const r1 = await client.sendRequest(
+        'cf_pool_orders',
+        { chain: 'Ethereum', asset: 'ETH' },
+        { asset: 'USDC', chain: 'Ethereum' },
+      );
+
+      const r2 = await client.sendRequest(
```

Comment:
why does this one not batch? because the request above took more than 100ms?

Reply:
yes, we awaited on the first one. so second one will run only once the first one was handled successfully after 100ms

Reply:
oh yeah this was a stupid question in hindsight lol